dna/models.py

```diff
@@ -445,20 +433,18 @@ def forward(self, args):
         assert len(metafeatures) == batch_size
         assert len(pipeline_structure) == seq_len
 
-        # Pass the metafeatures through the input layer
-        metafeatures = self._input_submodule(metafeatures)
-
-        # Add a dimension to the metafeatures so they can be concatenated across that dimension
-        metafeatures = metafeatures.unsqueeze(dim=0)
-
-        # Initialize the hidden state and cell state using a concatenation of the metafeatures
-        hidden_state = []
-        for i in range(self.hidden_state_dim0_size):
-            hidden_state.append(metafeatures)
-        hidden_state = torch.cat(hidden_state, dim=0)
+        hidden_state = torch.zeros(self.hidden_state_dim0_size, batch_size, self.hidden_state_size, device=self.device)
         cell_state = hidden_state
         lstm_start_state = (hidden_state, cell_state)
 
+
+        dag_rnn_output = self.parse_DAG(seq_len, pipelines, pipeline_structure, lstm_start_state)
+        linear_input = torch.cat((dag_rnn_output, metafeatures), dim=1)
+
+        linear_output = self._output_submodule(linear_input)
```

Comment:
Why are these variables called "linear"? This are hidden states of the network. Also, the output submodule is probably not linear either. Perhaps there is a better name.